2020-01-Winter/1_linked_lists/README.md

```diff
@@ -118,15 +159,50 @@ Go to [Top](#top)
 
 ### 2. SOLUTION 2 TODO :bug:
 
-Source: TODO :bug:
-
-#### Naive/Simple Solution
-
-TODO :bug:
-
-#### Optimal Solution
-
-TODO :bug:
+Source:https://www.geeksforgeeks.org/flatten-a-linked-list-with-next-and-child-pointers/
+
+#### Solution
+
+	public Node flatten(Node head) {
+		/*Base case*/
+		if (head == null) { 
+			return head; 
+		} 
+
+		/* Find tail node of first level linked list */
+		Node tail = head; 
+		while (tail.next != null) { 
+			tail = tail.next; 
+		} 
+
+		// One by one traverse through all nodes of first level 
+		// linked list till we reach the tail node 
+		Node current = head; 
+		while (current != tail) { 
+			
+
+			// If current node has a child 
+			if (current.child != null) { 
+				// then link child in between current and current.next
+				//store last link on next level as temp to connect to
+				//current.next
+				Node temp = current.child; 
+				while (temp.next != null) { 
+					temp = temp.next; 
+				} 
+				//break and create new links
+				temp.next = current.next;
+				current.next.prev = temp;
+				current.next = current.child; 
+				current.child.prev = current;
+				current.child = null;
+			} 
+
+			// Change current node 
+			current = current.next; 
+		} 
+		return head;
+	}
 
 #### Testing The Solutions OR Driver For Solution
 
```

Comment:
Add a link here to the Java file you wrote, or just remove this section altogether